metagraph/src/graph/representation/canonical_dbg.cpp

```diff
@@ -195,21 +197,33 @@ void CanonicalDBG::append_next_rc_nodes(node_index node,
 
                 c = kmer::KmerExtractorBOSS::complement(c);
 
-                if (children[c] == npos)
+                if (children[c] != npos) {
+                    if (!k_odd_) {
+                        is_palindrome_cache_.Put(next, true);
+                    } else {
+                        throw std::runtime_error(fmt::format(
+                            "Primary graph contains both forward and reverse complement: {} {} -> {} {}\t{} {}",
+                            node, graph_.get_node_sequence(node),
+                            children[c], graph_.get_node_sequence(children[c]),
+                            next, graph_.get_node_sequence(next)
+                        ));
+                    }
+
+                } else {
                     children[c] = next + offset_;
+                }
             },
             get_k() - 1
         );
 
     } else {
         for (size_t c = 0; c < alphabet.size(); ++c) {
-            if (children[c] != npos)
-                continue;
-
-            rev_seq[0] = complement(alphabet[c]);
-            node_index next = graph_.kmer_to_node(rev_seq);
-            if (next != npos)
-                children[c] = next + offset_;
+            if (alphabet[c] != boss::BOSS::kSentinel && children[c] == npos) {
+                rev_seq[0] = complement(alphabet[c]);
+                node_index next = graph_.kmer_to_node(rev_seq);
+                if (next != npos)
+                    children[c] = next + offset_;
+            }
```

Comment:
writing some short comments would be nice

Reply:
done